test/common/upstream/health_check_fuzz.cc

```diff
@@ -3,11 +3,92 @@
 #include <chrono>
 #include <memory>
 
+#include "common/grpc/common.h"
+
 #include "test/common/upstream/utility.h"
 #include "test/fuzz/utility.h"
 
 namespace Envoy {
 namespace Upstream {
+namespace { // gRPC helper methods
+// From unit tests
+std::vector<std::vector<uint8_t>>
+serializeResponseToBufferList(grpc::health::v1::HealthCheckResponse::ServingStatus status,
+                              uint64_t chunk_size_from_fuzzer) {
+  grpc::health::v1::HealthCheckResponse response;
+  response.set_status(status);
+  const auto data = Grpc::Common::serializeToGrpcFrame(response);
+  uint64_t chunk_size = chunk_size_from_fuzzer % data->length();
+  if (chunk_size == 0) {
+    ++chunk_size;
+  }
+  std::vector<std::vector<uint8_t>> bufferList;
+  for (size_t i = 0; i < data->length(); i += chunk_size) {
+    if (i >= data->length() - chunk_size) {
+      // The length of the last chunk
+      chunk_size = data->length() - i;
+    }
+    auto buffer = std::vector<uint8_t>(chunk_size, 0);
+    data->copyOut(i, chunk_size, &buffer[0]);
+    bufferList.push_back(buffer);
+  }
+  return bufferList;
+}
+
+grpc::health::v1::HealthCheckResponse::ServingStatus
+convertToGrpcServingStatus(test::common::upstream::ServingStatus status) {
+  switch (status) {
+  case test::common::upstream::ServingStatus::UNKNOWN: {
+    return grpc::health::v1::HealthCheckResponse::UNKNOWN;
+  }
+  case test::common::upstream::ServingStatus::SERVING: {
+    return grpc::health::v1::HealthCheckResponse::SERVING;
+  }
+  case test::common::upstream::ServingStatus::NOT_SERVING: {
+    return grpc::health::v1::HealthCheckResponse::NOT_SERVING;
+  }
+  case test::common::upstream::ServingStatus::SERVICE_UNKNOWN: {
+    return grpc::health::v1::HealthCheckResponse::SERVICE_UNKNOWN;
+  }
+  default: // shouldn't hit
+    NOT_REACHED_GCOVR_EXCL_LINE;
+  }
+}
+
+std::vector<std::vector<uint8_t>>
+makeBufferListToRespondWith(test::common::upstream::GrpcRespondBytes grpc_respond_bytes) {
+  std::vector<std::vector<uint8_t>> bufferList;
```

Comment:
I think you can move `bufferList` to the second case block.

Reply:
Done, I switched the return to that block too.